src/sumo/wrapper/_auth_provider.py

```diff
@@ -57,6 +57,9 @@ def get_token(self):
         return result["access_token"]
 
     def get_authorization(self):
+        if self.get_token() is None:
+            return ""
+
         return {"Authorization": "Bearer " + self.get_token()}
```

Comment:
Maybe do something like
````
token = self.get_token()
if token is None:
    return ""
return {"Authorization": "Bearer " + token}
```

The `.acquire_token_silent()` probably caches the token value, but we still avoid multiple calls to the token method.